tests/wasmldr_tests.rs

```diff
@@ -26,32 +26,46 @@ fn create(path: &Path) {
     }
 }
 
-pub fn enarx_run<'a>(wasm: &str, input: impl Into<Option<&'a [u8]>>) -> Output {
+pub fn enarx_run<'a>(
+    wasm: &str,
+    bin_args: impl Into<Option<&'a [&'a str]>>,
+    input: impl Into<Option<&'a [u8]>>,
+) -> Output {
+    let bin_args = bin_args.into();
+
     let wasm_path = Path::new(CRATE)
         .join(OUT_DIR)
         .join(TEST_BINS_OUT)
         .join(wasm);
 
-    let mut child = Command::new(&String::from(KEEP_BIN))
+    let mut child = Command::new(&KEEP_BIN);
+    let mut child = child
         .current_dir(CRATE)
         .arg("run")
         .arg(wasm_path)
         .stdin(Stdio::piped())
         .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
+        .stderr(Stdio::piped());
+
+    if let Some(args) = bin_args {
+        child = child.args(args);
+    }
+
+    let mut child = child
         .spawn()
         .unwrap_or_else(|e| panic!("failed to run `{}`: {:#?}", wasm, e));
 
-    if let Some(input) = input.into() {
-        child
-            .stdin
-            .as_mut()
-            .unwrap()
-            .write_all(input)
-            .expect("failed to write stdin to child");
-
-        drop(child.stdin.take());
-    }
+    let input_thread = if let Some(input) = input.into() {
+        let mut stdin = child.stdin.take().unwrap();
+        let input = input.to_vec();
+        Some(std::thread::spawn(move || {
```

Comment:
Why are we introducing threading here?

Reply:
Because otherwise we are blocked on writing the `input`, when the other end is waiting for us to read the `output`.